### ISSUE TITLE
Handle logits with more than 2 axes in `torch.func` diag and full backends

### ISSUE THREAD

**Issue by wiseodd** [aleximmer/Laplace#178]
Currently, we still assume that `logits.shape == (batch_size, n_classes)`. E.g.: https://github.com/aleximmer/Laplace/blob/a4d3ed631c87ddb55a1bc7aa053da249180480f7/laplace/curvature/curvature.py#L332-L334
